Foundation/NSThread.swift

```diff
@@ -57,18 +57,30 @@ private func NSThreadStart(_ context: UnsafeMutableRawPointer?) -> UnsafeMutable
 }
 
 open class Thread : NSObject {
-    
+    static internal let _mainThread = Thread(thread: pthread_self())
```

Comment:
I didn't find any api for this, only suggestions to use pthread_self in the main function.
And i found that Linux [kernel tests](https://github.com/torvalds/linux/blob/5924bbecd0267d87c24110cbe2041b5075173a25/tools/testing/selftests/mqueue/mq_perf_tests.c#L545) use this technique as well. 

I checked current implementation with a complete toolchains build and this code
```swift
import Foundation
Thread {
    print(Thread.main == Thread.current)
}.start()
sleep(1)
```
prints `false`, so the expected behaviour is correct.